cid.go

```diff
@@ -337,6 +337,19 @@ func (c *Cid) StringOfBase(base mbase.Encoding) (string, error) {
 	}
 }
 
+// Encode return the string representation of a Cid in a given base
+// when applicable
```

Comment:
I'd definitely expand on the edge cases here. Other than that, I have no real objections if we end up needing this.